sklearn/utils/estimator_checks.py

```diff
@@ -156,6 +157,29 @@ def check_supervised_y_no_nan(name, estimator_orig):
                          "array y with NaN value.".format(name))
 
 
+def check_constant_features(name, estimator_orig):
+    # Checks that estimators work with constant features
+    # or raise a reasonable error message
+    estimator = clone(estimator_orig)
+
+    # create features with a constant column
+    rng = np.random.RandomState(888)
+    if name == 'SVC':
```

Comment:
There really shouldn't be special cases. Why was that needed?

Reply:
I need it to avoid having the ValueError `X.shape[0] should be equal to X.shape[1]` while testing SVC. When the method fit() is called for  SVC, seems that the kernel is set by default to "precomputed" and this requires `X.shape[0] == X.shape[1]` (sklearn/svm/base.py, line 160)

Reply:
The kernel shouldn't be ``'precomputed'`` by default, it's ``'rbf'``

Reply:
You are right. However, after looking at the code I found that the test causing the fit of SVC is actually `test_check_estimator_pairwise`:
```
def test_check_estimator_pairwise():
    # check that check_estimator() works on estimator with _pairwise
    # kernel or  metric

    # test precomputed kernel
    est = SVC(kernel='precomputed')
    check_estimator(est)

    # test precomputed metric
    est = KNeighborsRegressor(metric='precomputed')
    check_estimator(est)
```

Reply:
Ah, right. You need to use ``pairwise_estimator_convert_X`` on ``X``.

Reply:
Thanks for the hint! I added it!